### PR TITLE
create checkbox component

### PR DESCRIPTION
**Pull request by hesamhabibih**
create checkbox component

### CODE REVIEW

File: components/CheckBox/CheckBox.type.tsx [hunk 1]
```diff
@@ -0,0 +1,14 @@
+export interface ICheckboxProps {
+  isChecked: boolean
+  isDisabled: boolean
+  isReadonly: boolean
+  label: string | undefined
```
Comment (amirkh330) [inline components/CheckBox/CheckBox.type.tsx]: `label?: string`
its better

File: components/CheckBox/CheckBox.type.tsx [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+  id: string | undefined
+  className?: string | undefined
+  name: string | undefined
+  direction?: 'flex | flex-col'
```
Comment (amirkh330) [inline components/CheckBox/CheckBox.type.tsx]: direction?: 'flex-row' | 'flex-col'
each of items must be in inside " ", and use flex-row

File: components/CheckBox/CheckBox.type.tsx [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+  backgroundColor: string | null
+  borderColor: string | null
+  textColor: string | null
+  onChange: (obj: any) => void
```
Comment (amirkh330) [inline components/CheckBox/CheckBox.type.tsx]: i think type of obj is `React.ChangeEvent<HTMLInputElement>` not any

File: components/CheckBox/CheckBox.tsx [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+  } = CheckboxProps
+
+  return (
+    <>
+      <div
+        className={`${backgroundColor} border ${borderColor} ${textColor} 
+      text-sm rounded-lg p-3 focus:ring-blue-500 focus:border-blue-500 block 
+      w-full dark:bg-gray-700 dark:border-gray-600 dark:placeholder-gray-400 
+      dark:text-white dark:focus:ring-blue-500 dark:focus:border-blue-500 
+      ${direction} ${backgroundColor}`}
```
Comment (amirkh330) [inline components/CheckBox/CheckBox.tsx]: backgroundColor is duplicate

File: components/CheckBox/CheckBox.tsx [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+        style={{ border: '2px dotted red', height: '50px' }}>
```
Comment (amirkh330) [inline components/CheckBox/CheckBox.tsx]: its necessary?

File: components/CheckBox/CheckBox.tsx [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+        <label htmlFor={label}>{label}</label>
+        <input
+          style={styles.checkbox}
```
Comment (amirkh330) [inline components/CheckBox/CheckBox.tsx]: we can use it in className as tailwind style

Comment (hesamhabibih) [reply components/CheckBox/CheckBox.tsx]: I edit my code

File: components/CheckBox/CheckBox.tsx [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+    backgroundColor = 'bg-gray-50',
+    borderColor = 'border-gray-300',
+    textColor = 'text-gray-900',
+    onChange,
+  } = CheckboxProps
+
+  return (
+    <>
+      <div
+        className={`${backgroundColor} border ${borderColor} ${textColor}
```
Comment (amirkh330) [inline components/CheckBox/CheckBox.tsx]: border must be border-2

File: components/CheckBox/CheckBox.type.tsx [hunk 8]
```diff
@@ -0,0 +1,16 @@
...
+  isChecked: boolean
+  isDisabled: boolean
+  isReadonly: boolean
+  label?: string | undefined
+  id: string | undefined
+  className?: string | undefined
+  name: string | undefined
+  direction?: 'flex' | 'flex-col'
+  backgroundColor: string | null
+  borderColor: string | null
```
Comment (amirkh330) [inline components/CheckBox/CheckBox.type.tsx]: why null it must be string

File: components/CheckBox/CheckBox.type.tsx [hunk 9]
```diff
@@ -0,0 +1,16 @@
+import { ChangeEvent } from 'react'
+
+export interface ICheckboxProps {
+  isChecked: boolean
+  isDisabled: boolean
+  isReadonly: boolean
+  label?: string | undefined
+  id: string | undefined
+  className?: string | undefined
```
Comment (amirkh330) [inline components/CheckBox/CheckBox.type.tsx]: we must change className to customCss

File: components/CheckBox/CheckBox.type.tsx [hunk 10]
```diff
@@ -0,0 +1,16 @@
+import { ChangeEvent } from 'react'
+
+export interface ICheckboxProps {
+  isChecked: boolean
+  isDisabled: boolean
+  isReadonly: boolean
+  label?: string | undefined
```
Comment (amirkh330) [inline components/CheckBox/CheckBox.type.tsx]: when you use the Question mark (?), you don't need to write undefined

File: components/CheckBox/CheckBox.tsx [hunk 11]
```diff
@@ -0,0 +1,49 @@
+import React from 'react'
+import { ICheckboxProps } from './CheckBox.type'
+
+const styles = {
```
Comment (amirkh330) [inline components/CheckBox/CheckBox.tsx]: remove this lines 
we use tailwind...

File: components/CheckBox/CheckBox.type.tsx (hunk 8 above)
Comment (hesamhabibih) [reply components/CheckBox/CheckBox.type.tsx]: changed to undefined

**Review by amirkh330 — APPROVED** [review]

**Review by saralabafi — APPROVED** [review]

**Pull request merged**